bin/pygrb/pycbc_pygrb_plot_coh_ifosnr

```diff
@@ -230,65 +141,158 @@ outdir = os.path.split(os.path.abspath(opts.output_file))[0]
-trig_data = load_data(trig_file, ifos, vetoes, opts, slide_id=opts.slide_id)
+# Load triggers/injections (apply reweighted SNR cut, not vetoes)
+trig_data = ppu.load_data(trig_file, ifos, data_tag='trigs',
+                          rw_snr_threshold=opts.newsnr_threshold,
+                          slide_id=opts.slide_id)
+inj_data = ppu.load_data(found_missed_file, ifos, data_tag='injs',
+                         rw_snr_threshold=opts.newsnr_threshold,
+                         slide_id=0)
+
+# Extract needed trigger properties and store them as dictionaries
+# Based on trial_dict: if vetoes were applied, trig_* are the veto survivors
+# Coherent SNR is always used
+x_key = 'network/coherent_snr'
+keys = [x_key]
+# Get parameters necessary for antenna responses
+keys += ['network/ra', 'network/dec', 'network/end_time_gc']
+# Get event_ids
+keys += [ifo+'/event_id' for ifo in ifos]
+keys += ['network/event_id']
+# Get single ifo SNR data
+keys += [ifo+'/snr' for ifo in ifos]
+# Get sigma for each ifo
+keys += [ifo+'/sigmasq' for ifo in ifos]
+found_trigs_slides = ppu.extract_trig_properties(
+    trial_dict,
+    trig_data,
+    slide_dict,
+    segment_dict,
+    keys
+)
+found_trigs = {}
+for key in keys:
+    found_trigs[key] = numpy.concatenate(
+       [found_trigs_slides[key][slide_id][:] for slide_id in slide_dict]
+    )
 
-# Extract (or initialize) injection data
-inj_data = load_data(found_file, ifos, vetoes, opts, injections=True, slide_id=0)
+# Complete the dictionary found_trigs
+# 1) Do not assume individual IFO and network event_ids are sorted the same way
+for ifo in ifos:
+    sorted_ifo_ids = numpy.array(
+        [
+            numpy.where(found_trigs[ifo+'/event_id'] == idx)[0][0]
+            for idx in found_trigs['network/event_id']
+        ]
+    )
+    for key in [ifo+'/snr', ifo+'/sigmasq']:
+        found_trigs[key] = found_trigs[key][sorted_ifo_ids]
+
+# 2) Get antenna response based parameters
+found_trigs['sigma_tot'] = numpy.zeros(len(found_trigs[x_key]))
+for ifo in ifos:
+    antenna = Detector(ifo)
+    ifo_f_resp = ppu.get_antenna_responses(
+        antenna,
+        found_trigs['network/ra'],
+        found_trigs['network/dec'],
+        found_trigs['network/end_time_gc']
+    )
+    # Get the average for f_resp_mean and calculate sigma_tot
+    found_trigs[ifo+'/f_resp_mean'] = ifo_f_resp.mean()
+    found_trigs['sigma_tot'] += found_trigs[ifo+'/sigmasq'] * ifo_f_resp
+
+# 3) Calculate the mean, max, and min sigmas
+for ifo in ifos:
+    try:
+        sigma_norm = found_trigs[ifo+'/sigmasq'] / found_trigs['sigma_tot']
+        found_trigs[ifo+'/sigma_mean'] = sigma_norm.mean()
+        if ifo == opts.ifo:
+            found_trigs['sigma_max'] = sigma_norm.max()
+            found_trigs['sigma_min'] = sigma_norm.min()
+    except ValueError:
+        found_trigs[ifo+'/sigma_mean'] = 0
+        if ifo == opts.ifo:
+            found_trigs['sigma_max'] = 0
+            found_trigs['sigma_min'] = 0
```

Comment:
If you prefer, I'm happy to initialise these quantities to 0 and change them only if found_trigs is not empty.

Reply:
Yes I prefer that, I think it makes the intent clearer.

Reply:
Actually, I just tested a couple things, and:
1. The first line (definition of `sigma_norm`) will handle the case of empty `found_trigs` just fine (it will produce another empty array) so you can move it outside the check block.
2. You can pass the argument `initial` to `min()` and `max()` to handle the case of empty inputs, though you need to be careful to not mess up the result when the input is *not* empty (i.e. it may not actually work).

Reply:
I opted for avoiding `initial` because it does this:
```
>>> numpy.max([-1], initial=0)
0
>>> numpy.min([1], initial=0)
0
>>> 
```
which is not the behaviour we want.